onmt/Translator.py

```diff
@@ -93,7 +94,8 @@ def translateBatch(self, batch, dataset):
         _, src_lengths = batch.src
         src = onmt.IO.make_features(batch, 'src')
         encStates, context = self.model.encoder(src, src_lengths)
-        decStates = self.model.init_decoder_state(context, encStates)
+        decStates = self.model.init_decoder_state(
+                src[:, :, 0].transpose(0, 1), context, encStates)
```

Comment:
This transpose is ugly, and is my fault. I implemented Transformer with batch_first indexing. 

Can you hide it in `init_decoder_state` of transformer? Just pass in `src`, and do the rest there until I fix it.

Reply:
Okay, will do.